code/game/atoms.dm

```diff
@@ -789,15 +789,15 @@ GLOBAL_LIST_EMPTY(blood_splatter_icons)
 		var/index = blood_splatter_index()
 		var/icon/blood_splatter_icon = GLOB.blood_splatter_icons[index]
 		if(!blood_splatter_icon)
-			blood_splatter_icon = icon(initial(icon), initial(icon_state), , 1)		//we only want to apply blood-splatters to the initial icon_state for each object
-			blood_splatter_icon.Blend("#fff", ICON_ADD) 			//fills the icon_state with white (except where it's transparent)
+			blood_splatter_icon = icon(initial(icon), initial(icon_state), frame = 1)		//we only want to apply blood-splatters to the initial icon_state for each object
```

Comment:
I was experimenting to see if there was a better way to handle the dupe code in `add_blood_overlay`. I found that if the `initial`s are removed here, then the baton and scythe `add_blood_overlay` procs are removed so they instead use this one, blood overlays still work fine for objects. I'm not really sure why the `initial`s are here in the first place, maybe I'm missing something.

It might break for items which can transform their icon state, like the baton, so code would need to be added for those. (I can't actually think of any other item besides the baton and scythe right now). But anyway, just an idea to play around with, code works fine as is.

Reply:
I had the exact same thoughts actually. I tried removing the `initial()`s entirely from the parent proc and just using that, but like you said it caused problems with other 'transforming' items like energy swords and riot shields:
![image](https://user-images.githubusercontent.com/57483089/134783866-5f815cdf-077f-444d-b663-8af6302ab3c2.png)

There's more items with just a single sprite than multiple of course, and those could be tweaked individually, but I figured it would be better to fix this bug in its own PR rather than bundling it into a small refactor which would probably need a testmerge, considering the whole fix bundle thing.